intermediate_source/_torch_export_nightly_tutorial.py

```diff
@@ -435,34 +435,45 @@ def suggested_fixes():
 #
 # Currently, the steps to register a custom op for use by ``torch.export`` are:
 #
-# - Define the custom op using ``torch.library`` (`reference <https://pytorch.org/docs/main/library.html>`__)
-#   as with any other custom op
+# - If you’re writing custom ops purely in Python, use torch.library.custom_op.
 
-from torch.library import Library, impl
+import torch.library
+import numpy as np
 
-m = Library("my_custom_library", "DEF")
-
-m.define("custom_op(Tensor input) -> Tensor")
-
-@impl(m, "custom_op", "CompositeExplicitAutograd")
-def custom_op(x):
-    print("custom_op called!")
-    return torch.relu(x)
+@torch.library.custom_op("mylib::sin", mutates_args=())
+def sin(x):
+    x_np = x.numpy()
+    y_np = np.sin(x_np)
+    return torch.from_numpy(y_np)
 
 ######################################################################
-# - Define a ``"Meta"`` implementation of the custom op that returns an empty
-#   tensor with the same shape as the expected output
+# - You will need to provide abstract implementation so that PT2 can trace through it.
 
-@impl(m, "custom_op", "Meta")
-def custom_op_meta(x):
+@torch.library.register_fake("mylib::sin")
+def _(x):
     return torch.empty_like(x)
 
+# - Sometimes, the custom op you are exporting has data-dependent output, meaning
+# we can't determine the shape of the output at compile time. In this case, you can do
+# following:
+@torch.library.register_fake("mylib::op_that_has_data_dependent")
```

Comment:
Any example is using this fake op op_that_has_data_dependent? Might worth adding one if not.